test/Tizen.NUI.Samples/Tizen.NUI.Samples/Samples/TouchGestureSample.cs

```diff
@@ -37,42 +47,49 @@ public void Activate()
             mGestureDetector = new GestureDetectorManager(backView, new MyGestureListener());
             backView.TouchEvent += OnBackTouchEvent;
 
-            window.Add(backView);
-            window.Add(frontView);
+            backView.Add(frontView);
+
+            root.Add(backView);
+            window.Add(root);
         }
 
         private bool OnFrontTouchEvent(object source, View.TouchEventArgs e)
         {
             Tizen.Log.Error("NUI", $"OnFrontTouchEvent {e.Touch.GetState(0)}\n");
-            return false;
+            return true;
         }
 
 
         private bool OnBackTouchEvent(object source, View.TouchEventArgs e)
         {
             Tizen.Log.Error("NUI", $"OnBackTouchEvent {e.Touch.GetState(0)}\n");
-            mGestureDetector.FeedTouchEvent(source, e);
+            mGestureDetector.FeedTouchEvent(source, e, this);
             return false;
         }
 
         class MyGestureListener : GestureDetectorManager.GestureListener
         {
-          public override void OnTap(object sender, TapGestureDetector.DetectedEventArgs e)
+          public override void OnTap(object sender, TapGestureDetector.DetectedEventArgs e, object userData)
```

Comment:
"object userData" looks not clear. I think if this is really needed data, declaring and defining the structure for it is required for user to add and use it easily. (it's like defining the EventArgs)
OR if this is not for user, then it would be better to hide it...